server/src/main/java/io/druid/segment/realtime/FireDepartment.java

```diff
@@ -92,10 +97,29 @@ public Plumber findPlumber()
     return ioConfig.getPlumberSchool().findPlumber(dataSchema, tuningConfig, metrics);
   }
 
+  public boolean checkFirehoseV2() {
+  return ioConfig.getFirehoseFactoryV2() != null;
+  }
+  
   public Firehose connect() throws IOException
   {
+  	try{
+  		log.info("connecting firehose [%s]", ioConfig);
+  	} catch (Exception e) {
+  		log.info("failed to connect firehose");
+  	}
     return ioConfig.getFirehoseFactory().connect(dataSchema.getParser());
```

Comment:
ops. sorry I didn't clean out my test code thoroughly. This connect() is supposed to throw exception and RealtimeManager will catch it, just like the old way